base/bitarray.jl

```diff
@@ -1913,3 +1913,8 @@ function read!(s::IO, B::BitArray)
 end
 
 sizeof(B::BitArray) = sizeof(B.chunks)
+
+function promote_rule(::Type{BitArray{N}},::Type{Array{T,N}}) where {T,N}
+    typejoin(Bool,T) == Any && return promote_rule(Bool,T)
+    Array{T,N}
+end
```

Comment:
```suggestion
function promote_rule(a::Type{BitArray{N}}, b::Type{Array{T, N}}) where {T, N}
    return el_same(promote_type(Bool, T), a, b)
end
```

Reply:
But perhaps a better option is relax the existing method:
```julia
promote_rule(a::Type{Array{T,n}}, b::Type{Array{S,n}}) where {T,n,S} = el_same(promote_type(T,S), a, b)
```

to accepting `::Type{<:AbstractArray{S,n}}` for `a` and `b`